### PR TITLE
KIALI-2840 Redesign Service Details

### PR DESCRIPTION
**Pull request by aljesusg**
JIRA: [KIALI-2840](https://issues.jboss.org/browse/KIALI-2840)

Design: https://github.com/kiali/kiali-design/issues/116

## Demo

redesign

cc @serenamarie125 @abonas

### CODE REVIEW

**Comment by aljesusg** [comment]
So much code addition from snaps :(

**Comment by lucasponce** [comment]
In theory we wanted to work on these redesigns when all PF3/PF4 is stable.

Code looks ok to me and I'm fine to merge it but just if this doesn't bring any regression and all PF4 related topics are more or less under control :-).

**Comment by hhovsepy** [comment]
Do not select any namespace.
Go to Service list.
Go to service details.
Click on "Show on graph"
"No namespace is selected"  in Graph.

**Comment by aljesusg** [comment]
Ok let me check this @hhovsepy Thanks

**Comment by aljesusg** [comment]
I check this @hhovsepy  and it seems that in master is not workign too. So I'll fix here

**Comment by aljesusg** [comment]
Please not squash the commits there are 2 commits of 2 JIRAs.
Please @hhovsepy @lucasponce review it.
Thanks !!!

**Comment by lucasponce** [comment]
@aljesusg I think https://issues.jboss.org/browse/KIALI-2886 has also broken other things
But I haven't checked master vs this PR (going to do this now)
When you click show on graph, the focus animation only happens when you have traffic (not in unused nodes) and it doesn't work with workloads/app details.

Going to re-build master and check if it works there, but I wanted to comment this here.

**Comment by aljesusg** [comment]
mmmm I only check the same behaviour that @hhovsepy commented and this happens in master too @lucasponce . If this behaviour is expected by the focus animation we are ok because I only moved the link I didn't change it .

**Comment by lucasponce** [comment]
Sorry, but in master works, but with this PR the show on graph feature seems broken.
So, we need to figure out first that.

File: src/components/Pf/PfTitle.tsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  name: string;
+  cytoscapeGraph: string;
+  graphType: string;
+  icon: JSX.Element;
+}
+
+class PfTitle extends React.Component<PfTitleProps, PfTitleState> {
+  constructor(props: PfTitleProps) {
+    super(props);
+    const namespaceRegex = /namespaces\/([a-z0-9\-]+)\/([a-z0-9\-]+)\/([a-z0-9\-]+)(\/([a-z0-9\-]+))?(\/([a-z0-9\-]+))?/;
```
Comment (lucasponce) [inline src/components/Pf/PfTitle.tsx]: Note that namespaces might have '.' character
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
I learnt that in the hard way,
I used a regexp (actually, this was recommended by k8s) in
https://github.com/kiali/kiali-ui/blob/master/src/components/IstioWizards/ThreeScaleIntegration.tsx#L62

**Comment by skondkar** [comment]
Tested PR and observed below two issues from design (https://github.com/kiali/kiali-design/issues/116)

1) Change Info tab label to Overview ( The tab name is still 'Info' )
2) Breadcrumb shouldn’t include the tab name (Breadcrumb includes the tab name like -     ServicesNamespace: bookinfoService: detailsService Info )

Tested service details in general..looks fine to me except above two points from design.

**Comment by aljesusg** [comment]
> Tested PR and observed below two issues from design ([kiali/kiali-design#116](https://github.com/kiali/kiali-design/issues/116))
> 
> 1. Change Info tab label to Overview ( The tab name is still 'Info' )
> 2. Breadcrumb shouldn’t include the tab name (Breadcrumb includes the tab name like -     ServicesNamespace: bookinfoService: detailsService Info )
> 
> Tested service details in general..looks fine to me except above two points from design.

Thanks !!! @skondkar I forgot them !!! I pushed this changes please can you check again ?

**Comment by skondkar** [comment]
Verified that service details displays Overview tab and breadcrumb does not show the tab names. Looks fine to me.

**Comment by abonas** [comment]
is this going to be applied to workload/app pages too?

**Comment by aljesusg** [comment]
Yes @abonas , service, workload and app

**Comment by abonas** [comment]
> Yes @abonas , service, workload and app

so this PR applies it to all?

**Comment by abonas** [comment]
the title talks about service details only

**Comment by aljesusg** [comment]
I am checking @abonas, the title is not entirely correct :( This apply the to the breadcumb and title in all pages, we use the same component for that, but I see that there is a workload info tab instead overview tab so I need another PR to do this changes :(. I'll ping you

**Comment by abonas** [comment]
np @aljesusg it is not urgent. thank you

**Pull request merged**